gravitee-node-monitoring/src/main/java/io/gravitee/node/monitoring/NodeMonitoringService.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.node.monitoring;
+
+import io.gravitee.common.utils.UUID;
+import io.gravitee.node.api.Monitoring;
+import io.gravitee.node.api.NodeMonitoringRepository;
+import io.reactivex.Flowable;
+import io.reactivex.Maybe;
+import io.reactivex.Single;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Component
+public class NodeMonitoringService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NodeMonitoringService.class);
+    private static final int CLEANUP_DELAY = 600000;
+
+    private final NodeMonitoringRepository repository;
+    private final Map<String, Monitoring> monitoringHolder;
+
+    public NodeMonitoringService(@Lazy NodeMonitoringRepository repository) {
+        this.repository = repository;
+        this.monitoringHolder = new ConcurrentHashMap<>();
+    }
+
+    public Single<Monitoring> createOrUpdate(Monitoring monitoring) {
+
+        if (repository == null) {
+            LOGGER.debug("There is nowhere to persist the monitoring data {}", monitoring);
+            return Single.just(monitoring);
+        }
+
+        final String monitoringKey = monitoring.getNodeId() + monitoring.getType();
+
+        Maybe<Monitoring> monitoringObs;
+
+        if (monitoringHolder.containsKey(monitoringKey)) {
```

Comment:
Can we add a small comment here just to explain why we are doing this (managing clustering, etc...)